testing/infra/terraform/modules/soaktest_workers/main.tf

```diff
@@ -0,0 +1,141 @@
+locals {
+  ssh_user_name      = "apmsoak"
+  executable_name    = "apmsoak"
+  service_name       = "apmsoak.service"
+  network_name       = "apmsoak-network-${terraform.workspace}"
+  remote_working_dir = "apmsoak/"
+  config_hash = sha256(join("_",
+    [
+      terraform.workspace,
+      var.apm_server_url,
+      var.apm_secret_token,
+      var.apm_loadgen_max_rate,
+      var.apm_loadgen_agents_replicas,
+      var.elastic_agent_version,
+      var.fleet_url,
+    ]
+  ))
+}
+
+resource "tls_private_key" "worker_login" {
```

Comment:
I didn't know this resource type existed. How would this work when / if we need to ssh to a remote machine and debug failing `apmsoak` systemd services?

Reply:
You can get the private key from the backend state and ssh.